tests/syslog/syslog_utils.py

```diff
@@ -0,0 +1,139 @@
+"""
+    Helpful utilities for writing tests for the syslog feature.
+"""
+import logging
+import time
+import os
+
+
+DUT_PCAP_FILEPATH = "/tmp/test_syslog_tcpdump_{vrf}.pcap"
+DOCKER_TMP_PATH = "/tmp/"
+
+
+def add_syslog_server(dut, syslog_server_ip, source=None, vrf=None, port=None):
+    """
+    Add syslog server
+
+    Args:
+        dut (SonicHost): The target device
+        syslog_server_ip (str): Syslog server address
+        source (str): Source ip address
+        vrf (str): Vrf device (default,mgmt,Vrf-data)
+        port (str): Server udp port
+
+    """
+    cmd_add_syslog_server = 'sudo config syslog add {} '.format(syslog_server_ip)
+    if source:
+        cmd_add_syslog_server = "{} --source {} ".format(cmd_add_syslog_server, source)
+    if vrf:
+        cmd_add_syslog_server = "{} --vrf {} ".format(cmd_add_syslog_server, vrf)
+    if port:
+        cmd_add_syslog_server = "{} --port {} ".format(cmd_add_syslog_server, port)
+    logging.debug("add_syslog_server command is: {}".format(cmd_add_syslog_server))
+    return dut.command(cmd_add_syslog_server, module_ignore_errors=True)
+
+
+def del_syslog_server(dut, syslog_server_ip):
+    """
+    Del syslog server
+
+    Args:
+        dut (SonicHost): The target device
+        syslog_server_ip (str): Syslog server ip
+    """
+    dut.command('sudo config syslog del {} '.format(syslog_server_ip))
+
+
+def create_vrf(dut, vrf):
+    """
+    Create Vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+    """
+    dut.command('sudo config vrf add {} '.format(vrf), module_async=True)
+
+
+def remove_vrf(dut, vrf):
+    """
+    Remove Vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+    """
+    return dut.command('sudo config vrf del {} '.format(vrf), module_ignore_errors=True)
+
+
+def bind_interface_to_vrf(dut, vrf, interface):
+    """
+    Bind interface to the specified vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+        interface (str): interface
+    """
+    dut.command('sudo config interface vrf bind {} {} '.format(interface, vrf))
+
+
+def replace_ip_neigh(dut, neighbour, neigh_mac_addr, dev):
```

Comment:
@JibinBao why not to add static neighbors like this:
```
cat ipv6_neigh.json
{
    "NEIGH": {
        "Vlan1000|200:200:200:200::5": {
            "neigh": "b8:ce:f6:56:da:14",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::4": {
            "neigh": "b8:ce:f6:56:da:07",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::7": {
            "neigh": "b8:ce:f6:56:da:16",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::6": {
            "neigh": "b8:ce:f6:56:da:15",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::1": {
            "neigh": "b8:ce:f6:56:da:01",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::3": {
            "neigh": "b8:ce:f6:56:da:06",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::2": {
            "neigh": "b8:ce:f6:56:da:02",
            "family": "IPv6"
        },
        "Vlan1000|200:200:200:200::8": {
            "neigh": "b8:ce:f6:56:da:18",
            "family": "IPv6"
        }
    }
}
sonic-cfggen -w -j ipv6_neigh.json
```

Reply:
I  think my method is ok. It can configure ipv4 and ipv6 neigh. It is simpler.

Reply:
@JibinBao with the method you used it adds neighbors in a failed state. From HW perspective they are not resolved. It is useless to have dynamic neighbor defined like that

Reply:
I think our test point is not "neighbors", it is just an assistant test tool. Actually, it works well with the "ip neigh", the packet can be sent successfully with the neighbors created by "ip neigh".  In addition, many community tests use "ip neigh" to add neighbors. So, I think my method is ok.
Thanks.